packages/ui-components/src/components/toggle-tip/toggle-tip.tsx

```diff
@@ -0,0 +1,87 @@
+import { autoPlacement, computePosition, ComputePositionConfig, Middleware, offset, shift } from '@floating-ui/dom';
+import { Component, Host, h, Prop, Element, State } from '@stencil/core';
+import { isEmpty } from 'lodash-es';
+import { EIconName } from '../icon/icon.types';
+import { DEFAULT_AUTOPLACEMENT_CONFIG, DEFAULT_OFFSET, DEFAULT_SHIFT_CONFIG, DEFAULT_TOGGLETIP_CONFIG } from './toggle-tip.config';
+import { EToggleTipPosition } from './toggle-tip.types';
+
+/**
+ * @part toggle-tip-container - The toggle tip container.
+ * @part toggle-tip-slot-content - The toggle tip slot content.
+ */
+@Component({
+	tag: 'kv-toggle-tip',
+	styleUrl: 'toggle-tip.scss',
+	shadow: true
+})
+export class KvToggleTip {
+	/** (required) Text of tooltip */
+	@Prop({ reflect: true }) text!: string;
+	/** (required) KvIcon icon*/
+	@Prop({ reflect: true }) icon: EIconName = EIconName.Info;
+	/** (optional) Position of tooltip */
+	@Prop({ reflect: true }) position?: EToggleTipPosition;
+	/** (optional) Array of allowed positions of tooltip (if defined the 'position' is ignored) */
+	@Prop({ reflect: true }) allowedPositions?: EToggleTipPosition[];
+	/** (optional) Object with tooltip options */
+	@Prop({ reflect: true }) options?: Partial<ComputePositionConfig>;
+
+	/** State to store toggletip state */
+	@State() toggleTipOpen: boolean = false;
+
+	/** The Host's element reference */
+	@Element() el: HTMLKvToggleTipElement;
+
+	componentDidRender() {
+		const toggleIcon = this.el.shadowRoot.querySelector('#toggle-tip-content');
+		const toggletip = this.el.shadowRoot.querySelector('#toggle-tip') as HTMLElement;
+
+		const position = isEmpty(this.allowedPositions) ? this.position : undefined;
+		const middleware = [offset(DEFAULT_OFFSET), shift(DEFAULT_SHIFT_CONFIG)];
+
+		if (isEmpty(position)) {
+			middleware.push(
+				autoPlacement({
+					...DEFAULT_AUTOPLACEMENT_CONFIG,
+					allowedPlacements: this.allowedPositions
+				})
+			);
+		}
+
+		const onButtonClick = () => {
+			this.toggleTipOpen = !this.toggleTipOpen;
+			if (this.toggleTipOpen) {
+				toggletip.style.display = 'inline-block';
+				this.updatePosition(toggleIcon, toggletip, position, middleware);
+			} else toggletip.style.display = '';
```

Comment:
we don't tipically use single line `if/else` statements, please add brackets to the `else` block